pay-queue/src/pay_queue/services/payment_reconciliations.py

```diff
@@ -395,6 +401,48 @@ def _csv_error_handling(row, error_msg: str, error_messages: List[Dict[str, any]
     error_messages.append({'error': error_msg, 'row': row})
 
 
+def _handle_credit_invoices_and_adjust_invoice_paid(row):
+    """Create CfsCreditInvoices and adjust the invoice paid amount."""
+    application_id = _get_row_value(row, Column.APP_ID)
+    cfs_identifier = _get_row_value(row, Column.SOURCE_TXN_NO)
+    if CfsCreditInvoices.find_by_application_id(application_id):
+        current_app.logger.warning(f'Credit invoices exists with application_id {application_id}.')
+        return
+    if not (credit := CreditModel.find_by_cfs_identifier(cfs_identifier=cfs_identifier, credit_memo=True)):
+        current_app.logger.warning(f'Credit with cfs_identifier {cfs_identifier} not found.')
+        return
+    invoice_number = _get_row_value(row, Column.TARGET_TXN_NO)
+    CfsCreditInvoices(
+        account_id=_get_payment_account(row).id,
+        amount_applied=Decimal(_get_row_value(row, Column.APP_AMOUNT)),
+        application_id=application_id,
+        cfs_account=_get_row_value(row, Column.CUSTOMER_ACC),
+        cfs_identifier=cfs_identifier,
+        created_on=datetime.strptime(_get_row_value(row, Column.APP_DATE), '%d-%b-%y'),
+        credit_id=credit.id,
+        invoice_amount=Decimal(_get_row_value(row, Column.TARGET_TXN_ORIGINAL)),
+        invoice_number=invoice_number
+    ).flush()
+
+    amount = CfsCreditInvoices.credit_for_invoice_number(invoice_number)
+    invoices = db.session.query(InvoiceModel) \
+        .join(InvoiceReferenceModel, InvoiceReferenceModel.invoice_id == InvoiceModel.id) \
```

Comment:
is the invoice reference status relevant / needed here?

Reply:
There wouldn't be a row without it being a completed invoice reference and we're only looking at PAID invoices

Reply:
put in two checks just in case